### PR TITLE
Modify arktos mizar deployment scripts

### PR DESCRIPTION
**Pull request by clu2xlu**

### CODE REVIEW

**Comment by vinaykul** [comment]
Do we still need this PR in its current form now that we have merged Amit's PR?

**Comment by clu2xlu** [comment]
> Do we still need this PR in its current form now that we have merged Amit's PR?

Yes. This change is based on Amit's PR

File: replace_containerd_start_arktos_api.sh [hunk 1]
```diff
@@ -0,0 +1,13 @@
+cd $HOME
+wget -qO- https://github.com/futurewei-cloud/containerd/releases/download/tenant-cni-args/containerd.zip | zcat > /tmp/containerd
+sudo chmod +x /tmp/containerd
+sudo systemctl stop containerd
+sudo mv /usr/bin/containerd /usr/bin/containerd.bak
+sudo mv /tmp/containerd /usr/bin/
+sudo systemctl restart containerd
+sudo systemctl start docker
+export CONTAINER_RUNTIME_ENDPOINT="containerRuntime,container,/run/containerd/containerd.sock"
+sleep 15
```
Comment (vinaykul) [inline replace_containerd_start_arktos_api.sh]: Why do we need 15 second sleep here?

File: replace_containerd_start_arktos_api.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+cd $HOME
+wget -qO- https://github.com/futurewei-cloud/containerd/releases/download/tenant-cni-args/containerd.zip | zcat > /tmp/containerd
+sudo chmod +x /tmp/containerd
+sudo systemctl stop containerd
+sudo mv /usr/bin/containerd /usr/bin/containerd.bak
+sudo mv /tmp/containerd /usr/bin/
+sudo systemctl restart containerd
+sudo systemctl start docker
```
Comment (vinaykul) [inline replace_containerd_start_arktos_api.sh:8]: Why is this needed?

Comment (clu2xlu) [reply replace_containerd_start_arktos_api.sh:8]: We need replace arktos containerd with a modified version in order for mizar to work. If we dont do this, arktos does not pass tenant info to mizar

File: replace_containerd_start_arktos_api.sh (hunk 1 above)
Comment (clu2xlu) [reply replace_containerd_start_arktos_api.sh]: Because it takes a while for containerd to restart

Comment (vinaykul) [reply replace_containerd_start_arktos_api.sh]: What happens if it took 16 seconds to restart?

File: replace_containerd_start_arktos_api.sh (hunk 2 above)
Comment (vinaykul) [reply replace_containerd_start_arktos_api.sh:8]: What does that statement do if docker is already running? Also, are we now not using containerd instead of docker?

**Comment by vinaykul** [comment]
+Yunwen to review this to see if we need to consider VM runtime as well in this workflow.

File: replace_containerd_start_arktos_api.sh (hunk 1 above)
Comment (yb01) [reply replace_containerd_start_arktos_api.sh]: agree with Vinay, probably a better way to handle this is to check containerd status to ensure it is up or timeout with failure

**Review by yb01 — APPROVED** [review]
/lgtm, from vm runtime's perspective.

File: replace_containerd_start_arktos_api.sh (hunk 1 above)
Comment (vinaykul) [reply replace_containerd_start_arktos_api.sh]: Thanks Yunwen. @clu2xlu We should avoid such arbitrary timeouts as it is going to fail at some point, leading to random/intermittent failures that are very hard to debug. Please fix this to check for desired status. For e.g. I did something similar for node ready in https://github.com/CentaurusInfra/mizar/blob/dev-next/kind-setup.sh#L116

**Review by vinaykul — CHANGES_REQUESTED** [review]
Please get rid of arbitrary sleeps in the code that are destined to introduce intermittent failures that are hard to debug.
